src/views/core/members/UpdateProfile.vue

```diff
@@ -0,0 +1,217 @@
+    }
+  },
+  methods: {
+    saveUser () {
+      this.isSaving = true
+      this.errors = {}
+
+      const body = _.pick(this.user, [
+        'id',
+        'first_name',
+        'last_name',
+        'username',
+        'gender',
+        'about_me',
+        'address',
+        'phone',
+        'date_of_birth'
+      ])
+
+      this.axios.put(this.services['core'] + '/members/me', body).then(() => {
+        this.isSaving = false
+
+        this.$root.showSuccess('User is saved.')
+
+        /* istanbul ignore next */
+        return this.$router.push({
+          name: 'oms.members.view',
+          params: { id: this.user.username || this.user.id }
+        })
+      }).catch((err) => {
+        this.isSaving = false
+
+        if (err.response.status === 422) { // validation errors
+          this.errors = err.response.data.errors
+          return this.$root.showError('Some of the user data is invalid.')
+        }
+
+        this.$root.showError('Could not save user', err)
+      })
+    },
+    askToJoinBody (bodyId) {
+      this.$buefy.dialog.prompt({
+        message: 'Join body',
+        inputAttrs: {
+          placeholder: 'Motivation',
+          required: false
+        },
+        onConfirm: (motivation) => this.joinBody(bodyId, motivation)
+      })
+    },
+    joinBody (bodyId, motivation) {
+      this.isLoading = true
+      this.axios.post(this.services['core'] + '/bodies/' + bodyId + '/join-requests', {
+        motivation
+      }).then(() => {
+        this.$root.showSuccess('Join request is sent.')
+        this.isLoading = false
+        this.isRequestingMembership = true
+      }).catch((err) => {
+        this.isLoading = false
+        if (err.response.status === 422) {
+          this.$root.showError('You\'ve already requested to join this body.')
+        } else {
+          this.$root.showError('Could not sent join request', err)
+        }
+      })
+    },
+    changeEmail (newEmail) {
+      if (['aegee.eu', 'aegee.org'].some(domain => newEmail.includes(domain))) {
+        return this.$root.showError('Cannot use @aegee.org or @aegee.eu emails.')
+      }
+
+      this.axios.put(this.services['core'] + '/members/me/email', { new_email: newEmail }).then(() => {
+        this.$root.showSuccess('An email change is triggered. Check your new inbox for a confirmation.')
+      }).catch((err) => {
+        this.$root.showError('Error changing user email', err)
+      })
+    }
+  },
+  computed: mapGetters({
+    services: 'services',
+    loggedIn: 'loggedIn',
+    loginUser: 'user',
+    isValid: 'isValid',
+    validationErrors: 'validationErrors'
+  }),
+  mounted () {
+    this.axios.get(this.services['core'] + '/members/me').then((response) => {
+      this.user = response.data.data
+
+      return this.axios.get(this.services['core'] + '/bodies')
+    }).then((response) => {
+      this.bodies = response.data.data.filter(b => ['antenna', 'contact antenna', 'contact'].includes(b.type))
```

Comment:
Maybe reference this back to the constant set in the other file? Instead of copying the values

Reply:
makes sense, refactored it and other comment